### PR TITLE
Standalone CARE build fixes on rzansel

### PR DESCRIPTION
**Pull request by adayton1**
* Use the cub target imported by RAJA if available
* Update the blueos host config
* Remove ambiguous operator== overload in host_device_ptr and just rely on the ones for chai::ManagedArray

### CODE REVIEW

File: src/care/host_device_ptr.h [hunk 1]
```diff
@@ -456,14 +456,6 @@ namespace care {
       CARE_HOST void move(ExecutionSpace space) {
          MA::move(chai::ExecutionSpace((int) space));
       }
-      
-      CARE_HOST_DEVICE inline bool operator ==(host_device_ptr<T, Accessor> const & right) const {
```
Comment (adayton1) [inline src/care/host_device_ptr.h:460]: Alternatively, I could move this outside the class and add nullptr overloads.

Comment (adayton1) [reply src/care/host_device_ptr.h:460]: The CHAI version only compares the active pointers, but I think we enforce consistency enough that that should be sufficient. And most often we just compare against nullptr.

Comment (adayton1) [reply src/care/host_device_ptr.h:460]: To be honest, I'm not convinced either ours or CHAI's implementation is really correct. Perhaps we ought to be comparing the pointer records.

Comment (dtaller) [reply src/care/host_device_ptr.h:460]: I'll approve this MR in any case, but yes something to discuss with in the future

**Review by dtaller — APPROVED** [review]

**Review by liu15 — APPROVED** [review]

**Review by agcapps — APPROVED** [review]
Thanks, @adayton1 .

**Pull request merged**